src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -156,5 +152,21 @@ public static partial class JsonSerializer
 #endif
             }
         }
+
+        private static void WriteExtensionData(Utf8JsonWriter writer, ref WriteStackFrame frame)
+        {
+            DictionaryEntry entry = ((IDictionaryEnumerator)frame.Enumerator).Entry;
+            if (entry.Value is JsonElement element)
+            {
+                Debug.Assert(entry.Key is string);
+
+                string propertyName = (string)entry.Key;
+                element.WriteAsProperty(propertyName, writer);
```

Comment:
Similarly, I think we may need the explicit call to `AsSpan()` (since the span to string implicit converter is on core only)

We should just add the `WriteAsProperty` string-based overload to JsonElement to avoid this concern all together.

> System\Text\Json\Serialization\JsonSerializer.Write.HandleDictionary.cs(164,41): error CS1503: Argument 1: cannot convert from 'string' to 'System.ReadOnlySpan<char>'

Reply:
Did you start a separate discussion to add the public API?

For now, I'll just do the `AsSpan()`. When the overload is added we will need to check for all usages and clean up as appropriate.

Reply:
> Did you start a separate discussion to add the public API?

Yes. Added the overload. See: https://github.com/dotnet/corefx/pull/37789